### DISCUSSION TITLE
Error getting data from an APP with mitmproxy and wireguard

### DISCUSSION THREAD

**Discussion by TheHypnoo** [mitmproxy/mitmproxy#5753] [Q&A]
Hi all, I've been trying for a while to get what data it sends and where it sends it from an APP. In this case, I started using PCAPDroid to get the data, but I couldn't find a way, using your guide, from there I got to mitmproxy, where I first started using mitmproxy with the transparent proxy, which didn't work on Mac. From there I used mitmproxy v9.0, on Windows, Mac and Linux with the transparent mode and the new recommended mode, Wireguard. 

I still can't get the data. 

In this case I use an android Mobile with version 11.
With root, using the mitmproxy certificate added to the system, added Magisk modules: Always Trust User Certificates. LSPosed: JustTrustMe, SSLUnpinning.

I show images of the error that mitmproxy shows me.
<img width="983" alt="image" src="https://user-images.githubusercontent.com/31130069/203522791-97443e52-40cf-469a-aacc-8e32868dac93.png">

I would appreciate any help, advice or ideas.

**Comment by mhils** [comment]
Two issues here:

1. The first error indicates that mitmproxy does not trust the server certificate, which may be self-signed.
2. Errors 2-3 indicate that your client is still not trusting the mitmproxy CA, at least for some connections. https://docs.mitmproxy.org/dev/concepts-certificates/#certificate-pinning

**Reply by TheHypnoo** [reply]
@mhils 

What does it mean that mitmproxy doesn't trust the self signed certificate from the server, and how can I fix that part?

The second errors, I'm using an android phone and supposedly the APK has the certificates removed, and I'm using different ways to bypass the certification validation.

In this case, I'm going to try Frida (objection)

**Comment by 3052** [comment]
I found recently that MITM Proxy is failing with Frida, but if you use the same exact script with HTTP Toolkit it works. I prefer MITM Proxy, but if it doesn't work I have no choice but to use something else

**Reply by VanshAnakin** [reply]
What else did u use?
